tests/checkpointing/test_model_checkpoint.py

```diff
@@ -986,8 +986,8 @@ def assert_checkpoint_log_dir(idx):
 
         trainer.test(model)
         # resume_from_checkpoint is resumed when calling `.fit`
-        assert trainer.global_step == 0
-        assert trainer.current_epoch == 0
+        assert trainer.global_step == 4
```

Comment:
Any idea why this is different ?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/9413#discussion_r707093777
as you suggested.

Reply:
if we are now restoring also the state when testing/validating, this will be a breaking change. Let's make sure this is documented in the changelog and the PR marked as breaking change.

Reply:
```suggestion
        assert trainer.global_step == epochs * limit_train_batches
```